modelexpress_client/python/modelexpress/metadata/publisher.py

```
@@ -159,6 +203,40 @@ def _mark_stale(self) -> None:
+                # that is the case worth demoting.
+                return
+            self._unhealthy = True
+            reason = getattr(self._nixl_manager, "data_plane_error", None)
+            try:
+                from .. import p2p_pb2
+                self._update_status(p2p_pb2.SOURCE_STATUS_STALE)
+                logger.warning(
+                    f"[Worker {self._worker_rank}] NIXL agent unhealthy, marked "
+                    f"STALE so targets stop selecting it: {reason}"
+                )
+            except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)